sensor/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,153 @@
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu3.jpg"  alt="menu3">
+</p>
+
+## Calibration
+
+The calibration starts automatically after clicking the `Calibrate` button. It will keep calibrating the lidartag detections and apriltag detections until the number of the detections fits the user-defined `calibration_pairs` in the `Launcher configuration`.
+
+When the user starts the calibration, `rviz` and the `image view` should be displayed like below.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization1.jpg"  alt="visualization1">
+</p>
+
+After the tools detect the lidartag and apriltag, it will show the detection markers on the `rviz` and the `image view`. The text in rviz will also display the number of pairs of lidartag/apriltag detections.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization2.jpg"  alt="visualization2">
+</p>
+
+Once the user observes the number of pairs increases in the `rviz`, he can start moving the tag to another position.
+
+While moving the tag to another position, the user should move the tag decisively from one place to the other, not stopping and starting mid-way.
+Please make sure the new position is at least `calibration_min_pair_distance` from the closest detection in the set of converged detections and the tag is in the FOV of both lidar and camera.
+
+Depending on the mount and the ground surface, the tags can oscillate quite a bit, which can be detrimental to the calibration process. In this case, the user can stop the tag with its hand.
+
+At the end of the calibration, we can get 8 detection pairs which are shown below.
+
+![tag_based_pnp_calibrator](../images/tag_based_pnp_calibrator/visualization3.jpg)
+
+The output in the console should be as follows:
+
+```text
+[tag_based_pnp_calibrator_node]:  Initial reprojection error=15.41
+[tag_based_pnp_calibrator_node]:  Current reprojection error=1.09
+[tag_based_pnp_calibrator_node]:  Filtered reprojection error=1.10
+```
+
+## Results
+
+After the calibration process finishes, the `sensor_calibration_manager` will display the results in the UI and allow the user to save the calibration data to a file.
+
+In the UI of the X2 project, three different TF trees are displayed: `Initial TF Tree`, `Calibration Tree`, and `Final TF Tree`. The `Initial TF Tree` presents the initial TF connections between sensors needed for calibration. The `Calibration Tree` shows the calibrated transformation between sensors, in this tutorial, `camera6/camera_optical_link` and `pandar_40p_front`. The `Final TF Tree` depicts the TF tree after incorporating the updated calibrated transformation. Since the transformations represented by the black arrows are fixed, the transformation between `front_unit_base_link` and `camera6/camera_link`, which is represented by the red arrow, can be calculated using the calibrated transformation.
```

Comment:
The documentation for `Final TF tree` is not quite correct

Reply:
"Since the transformations represented by the black arrows are fixed, the transformation between front_unit_base_link and camera6/camera_link, which is represented by the red arrow, can be calculated using the calibrated transformation."

is also kind of wrong. Please redact it again. For example it makes the user think black = fixed

Reply:
@knzo25 
sry, could you explain them correctly, thanks.

I mentioned the black arrows (transformations) are fixed.

Reply:
> I mentioned the black arrows (transformations) are fixed.
Since in the figure are several black arrows that are not necessarily fixed you need to specify.
Also, you imply that because they are black they are fixes. That is not the case. The calibrator interface (.py file) determined as so. Why? That needs to be explained or it makes no sense. The color code actually means whether or not the connection changed. Please careful that you understand with what you write and that that is correct

Reply:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/e2210ae2139ca0627ac335674fa091113f8bee0c

Reply:
@vividf 
The explanation itself id good this time. However, the redaction is a little off so I will make a separate comment.